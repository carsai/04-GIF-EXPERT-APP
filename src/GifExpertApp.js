import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { AddCategory } from './components/AddCategory';
import { GifGrip } from './components/GifGrip';

export const GifExpertApp = ({defaultCategoria=[]}) => {

    const [categorias, setCategorias] = useState(defaultCategoria);

    return (
        <>
            <h2>GifExperApp</h2>

            <AddCategory setCategorias={setCategorias} />

            <hr/>

            <ol>
                {
                    categorias.map( c => 
                            <GifGrip 
                                category={c} 
                                key={c}
                            />
                        )
                }
            </ol>
        </>
    );
}

GifExpertApp.propTypes = {
    defaultCategoria: PropTypes.array
}