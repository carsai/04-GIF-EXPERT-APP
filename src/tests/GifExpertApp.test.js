import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas con <GifExpertApp />', () => {
    
    test('Debe mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de categorias', () => {

        const categorias = ['One Punch', 'Goku'];

        const wrapper = shallow(<GifExpertApp defaultCategoria={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrip').length).toBe(categorias.length);
        
    });
    
    
});
