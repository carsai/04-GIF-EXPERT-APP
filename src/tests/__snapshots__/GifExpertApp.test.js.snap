// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrip
      category="One Punch"
      key="One Punch"
    />
    <GifGrip
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
